wgengine/magicsock/magicsock.go

```diff
@@ -3008,3 +3008,13 @@ func getPeerMTUsProbedMetric(mtu tstun.WireMTU) *clientmetric.Metric {
 	mm, _ := metricRecvDiscoPeerMTUProbesByMTU.LoadOrInit(key, func() *clientmetric.Metric { return clientmetric.NewCounter(key) })
 	return mm
 }
+
+// GetLastNetcheckReport returns the last netcheck report.
+func (c *Conn) GetLastNetcheckReport(ctx context.Context) *netcheck.Report {
+	nr, err := c.updateNetInfo(ctx)
```

Comment:
This triggers a new netcheck, is there a way to look up the results of the last one that already finished? Then we can only trigger a new one if there's no previous report.